### PR TITLE
[main] Updated issues template

### LINKED ISSUE #677: Update the issues template to help users identify which extension they are using
Since we now support a `server` extension, it would be beneficial to the project to have an issue template with the following fields:

**Extension** (checkbox style input): `server` or `client` (here we should be able to add a correct label to the issue)
**Java version**: output of java --version
**Quarkus version**: Platform Quarkus version used in the project

And as usual, **Description** and **Reproducer** in case of bugs.

### PR DESCRIPTION
**Pull request by gmalheiro**
Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [x] Pull Request contains link to the issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

---
## Issue description:
Updated the issue template to improve issue categorization and handling by adding specific fields and clarifications. This update ensures that all necessary information is captured right at the start, aiding in more efficient issue resolution.

Fixes #677

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+name: "[Issue number - problem description]"
+about: This template standardizes issue reporting within the project.
+title: "[Issue]"
+labels: bug, documentation, duplicate, enhancement, good first issue, help wanted,
+  invalid, question, wontfix
+assignees: gmalheiro
```
Comment (ricardozanini) [inline .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:7]: Can you please remove this line?

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+name: "[Issue number - problem description]"
+about: This template standardizes issue reporting within the project.
+title: "[Issue]"
+labels: bug, documentation, duplicate, enhancement, good first issue, help wanted,
+  invalid, question, wontfix
```
Comment (ricardozanini) [inline .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:6]: So if I open an issue, all these labels are added? I think we should rethink this. The labels will be added based on the checkbox in the `Extension` section. And for that, we will have to create a bot. 😅 

Unless there's an option for users to pick the area once open an issue here. For example, the flow can be something like: Open an Issue, Pick from `server`, `client`, or `general`. So then GH can assign the correct labels.

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md (hunk 1 above)
Comment (gmalheiro) [reply .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:7]: Sure!!

**Comment by gmalheiro** [comment]
@ricardozanini I've removed the lines you requested. Thanks for the feedback! Could you please confirm if everything is as expected?

**Review by hbelmiro — COMMENTED** [review]
The assignee is still there @gmalheiro.

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+## Extension
+- [ ] Server
+- [ ] Client
+
+## Java Version
+Please provide the output of `java --version`.
+
+## Quarkus Version
+Please specify the version of Quarkus used.
```
Comment (mcruzdev) [inline .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:14]:
```suggestion
Please, specify the version of Quarkus used.
```

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+name: "[Issue number - problem description]"
+assignees: gmalheiro
+---
+
+## Extension
+- [ ] Server
+- [ ] Client
+
+## Java Version
+Please provide the output of `java --version`.
```
Comment (mcruzdev) [inline .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:11]:
```suggestion
Please, provide the output of `java --version`.
```

**Review by mcruzdev — COMMENTED** [review]
Looks good @gmalheiro!

I left some comments!

**Comment by gmalheiro** [comment]
@mcruzdev Thanks for the suggestions! I accepted them!

@hbelmiro I removed the assignees section! Thanks for the feedback.

**Review by hbelmiro — APPROVED** [review]
Thank you @gmalheiro.

**Review by mcruzdev — APPROVED** [review]

**Comment by mcruzdev** [comment]
Thank you @gmalheiro, could you squash the commit?

**Comment by gmalheiro** [comment]
@mcruzdev Done! Thanks for the support! 😁

**Review by ricardozanini — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+---
+name: "[Issue number - problem description]"
+---
+
+## Extension
+- [ ] Server
+- [ ] Client
+
+## Java Version
+Please, provide the output of `java --version`.
```
Comment (ricardozanini) [inline .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:10]:
```suggestion
<!-- Please, provide the output of `java --version`. -->
```

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+## Quarkus Version
+Please, specify the version of Quarkus used.
```
Comment (ricardozanini) [inline .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:13]:
```suggestion
<!-- Please, specify the version of Quarkus used. -->
```

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+## Description
+Please, provide a detailed description of the issue or the feature request.
```
Comment (ricardozanini) [inline .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:16]:
```suggestion
<!-- Please, provide a detailed description of the issue or the feature request. -->
```

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+## Steps to Reproduce (if applicable)
+Please, provide detailed steps to reproduce the bug, including any relevant code snippets or links to files.
```
Comment (ricardozanini) [inline .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:19]:
```suggestion
<!-- Please, provide detailed steps to reproduce the bug, including any relevant code snippets or links to files. -->
```

**Review by ricardozanini — COMMENTED** [review]
By adding a comment to the descriptions, we hide this information in the rendered issue.

**Comment by gmalheiro** [comment]
@ricardozanini  Tks for the suggestions! I accepted it!

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md (hunk 7 above)
Comment (hbelmiro) [reply .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:16]: @gmalheiro it doesn't seem like you accepted this one.

**Review by mcruzdev — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md (hunk 7 above)
Comment (hbelmiro) [reply .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:16]: @gmalheiro can you please comment this line before we get it merged?

**Review by hbelmiro — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md (hunk 7 above)
Comment (gmalheiro) [reply .github/ISSUE_TEMPLATE/-issue-number---problem-description-.md:16]: @hbelmiro  Sure!

**Review by hbelmiro — APPROVED** [review]

**Comment by hbelmiro** [comment]
@all-contributors add @gmalheiro for docs.

**Pull request merged**